workflow-generator/src/main/java/com/wipro/ats/bdre/wgen/RegisterPartitionsActionNode.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.wipro.ats.bdre.wgen;
+
+import com.wipro.ats.bdre.md.api.GetProperties;
+import com.wipro.ats.bdre.md.beans.ProcessInfo;
+import org.apache.log4j.Logger;
+
+import java.util.Enumeration;
+
+/**
+ * Created by cloudera on 3/31/16.
+ */
+
+
+/*
+Action nodes are the mechanism by which a workflow triggers the execution of a task
+Here, we set the id and return name of the action node.
+The method getXML() returns a string which contains name, Id, next success node(ToNode) and next failure node(TermNode)
+for the current action node, appropriately formatted as XML.
+*/
+
+public class RegisterPartitionsActionNode extends GenericActionNode {
+
+    private static final Logger LOGGER = Logger.getLogger(RegisterPartitionsActionNode.class);
+    private ProcessInfo processInfo = new ProcessInfo();
+    private ActionNode actionNode = null;
+
+    /**
+     * This constructor is used to set node id and process information.
+     *
+     * @param actionNode An instance of ActionNode class which a workflow triggers the execution of a task.
+     */
+    public RegisterPartitionsActionNode(ActionNode actionNode) {
+        setId(actionNode.getId());
+        processInfo = actionNode.getProcessInfo();
+        this.actionNode = actionNode;
+    }
+
+    public ProcessInfo getProcessInfo() {
+        return processInfo;
+    }
+
+
+    public String getName() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1161)